### PR TITLE
[prometheus-adapter] Allow namespace override

### LINKED ISSUE #1931: [prometheus-adapter] Add namespaceOverride option
### Is your feature request related to a problem ?

We are installing prometheus-adapter as a dependency of our chart, and we want to specify a custom namespace for it.

### Describe the solution you'd like.

Add a namespaceOverride option similar to what kube-prometheus-stack chart has.

### Describe alternatives you've considered.

NONE

### Additional context.

Thank you!

### PR DESCRIPTION
**Pull request by gshaibi**
#### What this PR does / why we need it:

Allow the release namespace to be overridden for multi-namespace deployments in combined charts

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #1931

#### Special notes for your reviewer:
Thanks!

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by monotek — CHANGES_REQUESTED** [review]
Please rebase. that should fix the linter.

**Comment by gshaibi** [comment]
Rebased @monotek

**Review by monotek — APPROVED** [review]

File: charts/prometheus-adapter/templates/_helpers.tpl [hunk 1]
```diff
@@ -24,6 +24,13 @@ If release name contains chart name it will be used as a full name.
 {{- end -}}
 {{- end -}}
 
+{{/*
+Allow the release namespace to be overridden for multi-namespace deployments in combined charts
+*/}}
+{{- define "k8s-prometheus-adapter.namespace" -}}
+{{- default .Release.Namespace .Values.namespaceOverride -}}
```
Comment (steven-sheehy) [inline charts/prometheus-adapter/templates/_helpers.tpl:31]: We should add `namespaceOverride: ""` to `values.yaml` and document it.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: charts/prometheus-adapter/templates/_helpers.tpl (hunk 1 above)
Comment (gshaibi) [reply charts/prometheus-adapter/templates/_helpers.tpl:31]: Added

**Review by monotek — CHANGES_REQUESTED** [review]
Please sign commits to get doc check pass.

**Comment by gshaibi** [comment]
Signed. thanks!

**Review by monotek — APPROVED** [review]

**Review by steven-sheehy — APPROVED** [review]

**Pull request merged**